Add post about another improving reponsiveness sprint (#246)
openSUSE/obs-landing: _posts/development/2020-03-03-who-said-obs-was-unusable.md

---
layout: post

> title: Who Said OBS Was Unusable on Mobile Devices?

Author: What about replacing the title with "Who Said OBS Was Not Responsive?"?

Any other suggestion?

> title: Who Said OBS Was Unusable on Mobile Devices?

Reviewer: Go for it!

> title: Who Said OBS Was Unusable on Mobile Devices?

Author: Done.

category: development
---

If you believe so, we're hopefully going to change your opinion next time you surf OBS within our [beta program](https://openbuildservice.org/beta/).

> In the last two weeks, we continued working in improving responsiveness, following a mobile-first approach.

Reviewer 2: what about `... kept working in ...`? `continued` sounds weird to me.

Author: Done.

Lots of issues were fixed and a bunch of changes were made.

## What's New?

> We have concentrated on specific components, most of them widely used in several parts of the application:

Reviewer 2: :thought_balloon: I would use `focused` instead of `concentrated`

Author: Done. Please re-review.

### Datatables

We changed how the search box and the pagination look like, mainly to have them fit on smaller screens.
We also fixed some issues when switching tabs on the page of a user's tasks, where the tables were overflowing the main container.

<img src="/images/posts/who-said-obs-was-unusable/datatables-search-box.png" alt="Datatables Search Box" style="width: 30%;" class="left" />
<img src="/images/posts/who-said-obs-was-unusable/datatables-pagination.png" alt="Datatables Pagination" style="width: 30%;" class="right" />
<img src="/images/posts/who-said-obs-was-unusable/datatables-fix-tab.png" alt="Datatables Fix Tab" style="width: 30%;" class="right" />

> 

Author: I removed the navigation bars and added red arrows to make it easier to recognize the named changes. Please re-review.

> There are still other datatables that we have to look into from the "mobile first" perspective to standardize them, thus having a similar behavior, look and feel for all tables.

Author: Changed.

### Build Results

We re-thinked the buildresults and added some features, and also fixed some issues.
One of the more relevant features is that we made the refresh button sticky to the buildresult.
So for extensive buildresult list, you don't need any more to go up to refresh the view.
This button was also moved to the right side to make it more thumb-friendly in mobile devices.

<img src="/images/posts/who-said-obs-was-unusable/sticky-refresh-button.gif" alt="Sticky Refresh Button" width="400" />

> Another feature that is closely related to the previous is that now if you have multibuild, you can collapse them and focus only on the package you are interested in.

Reviewer 3: Would write it a bit different, a suggestion would be:

Another feature, closely related to the previous one, enables you to collapse the build results of multibuild targets.
This way you can focus only on the results of packages you are interested in.

Author: Changed. Please re-review.

<img src="/images/posts/who-said-obs-was-unusable/multibuild-collapsible.gif" alt="Multibuild Collapsible" width="400" />

> And last but not least, tooltips where removed in favor of a bigger help content section, showing all the help information for a particular build result.

Author: Ops, fixed.

Creating and displaying data is going to be on a page instead of a modal.
We will favor inline editing whenever possible instead of having a modal for editing data.
This all boils down to improve the experience of users on devices with smaller viewports.
Modals, especially those for creating and editing data, are often completely filling up the screens of such devices.
This is what we want to avoid.

This was the case of showing the GPG Key and the SSL Certificate of a project:

<img src="/images/posts/who-said-obs-was-unusable/gpg-key-and-ssl-certificate.png" alt="GPG Key and SSL Certificate" width="400" />

### Navigation

We did an extensive research regarding navigation components (navigation bars, tabs, breadcrumbs, contextual links, ...).
Stay tuned.
In the next weeks, we will provide a nice and coherent solution to make it easier to view and navigate through what matters to you.

## Try It out and Give Us Some Feedback

This time, we received fast feedback about some changes made in the beta program regarding improvement of responsiveness.
This allowed us to fix some issues before the majority of users noticed.

> We appreciate that and we thank you!

Reviewer: Why not list the issues? :-)

Author: Added.

Reviewer: You should add ad CR after the list or it looks funny...

Author: Argh. Thanks. Done.

In the last two weeks, we kept working in improving responsiveness following a mobile-first approach.
Lots of issues were fixed and a bunch of changes were made.

## What's New?

We have focused on specific components, most of them widely used in several parts of the application:

### Datatables

We changed how the search box and the pagination look like, mainly to have them fit on smaller screens.
We also fixed some issues when switching tabs on the page of a user's tasks, where the tables were overflowing the main container.

<img src="/images/posts/who-said-obs-was-unusable/datatables-search-box.png" alt="Datatables Search Box" style="width: 30%;" class="left" />
<img src="/images/posts/who-said-obs-was-unusable/datatables-pagination.png" alt="Datatables Pagination" style="width: 30%;" class="right" />
<img src="/images/posts/who-said-obs-was-unusable/datatables-fix-tab.png" alt="Datatables Fix Tab" style="width: 30%;" class="right" />

There are still other tables that we have to look into for different resolutions.
In the end we want to have a shared look and feel for all of them.

### Build Results

> We re-thinked the build results and added some features, and also fixed some issues.

Reviewer: ```suggestion
We have been rethinking the build results, added some features and also fixed some issues.
```

Author: Ops. Changed.
